mongo/mgo_job_store.go

```diff
@@ -0,0 +1,198 @@
+package mongo
+
+import (
+	"context"
+	"errors"
+	dpMongodb "github.com/ONSdigital/dp-mongodb"
+	dpMongoLock "github.com/ONSdigital/dp-mongodb/dplock"
+	dpMongoHealth "github.com/ONSdigital/dp-mongodb/health"
+	"github.com/ONSdigital/dp-search-reindex-api/models"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+)
+
+// MgoJobStore defines the required methods from MongoDB
+type MgoJobStore interface {
+	Close(ctx context.Context) error
+	CreateJob(ctx context.Context, id string) (job models.Job, err error)
+	GetJob(ctx context.Context, id string) (job models.Job, err error)
+	GetJobs(ctx context.Context) (job models.Jobs, err error)
+	AcquireJobLock(ctx context.Context, id string) (lockID string, err error)
+	UnlockJob(lockID string) error
+}
+
+//LastUpdatedSlice is a type that implements the sort interface so that the jobs in it can be sorted using the generic Sort function.
+type LastUpdatedSlice []models.Job
+
+//Len is a function that's required by the sort interface.
+func (s LastUpdatedSlice) Len() int {
+	return len(s)
+}
+
+//Less is a function that's required by the sort interface.
+func (s LastUpdatedSlice) Less(i, j int) bool {
+	return s[i].LastUpdated.Before(s[j].LastUpdated)
+}
+
+//Swap is a function that's required by the sort interface.
+func (s LastUpdatedSlice) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+// jobs collection name
+const jobsCol = "jobs"
+
+// locked jobs collection name
+const jobsLockCol = "jobs_locks"
+
+//MgoDataStore is a type that contains an implementation of the MgoJobStore interface, which can be used for creating and getting Job resources.
+//It also represents a simplistic MongoDB configuration, with session, health and lock clients
+type MgoDataStore struct {
```

Comment:
Mgo not needed, also I'd be more specific about the kind of data being stored.
```suggestion
type JobStore struct {
```

Reply:
done